src/mcp/tools/apphosting/list_backends.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { toContent } from "../../util.js";
+import { NO_PROJECT_ERROR } from "../../errors.js";
+import {
+  Backend,
+  getTraffic,
+  listBackends,
+  parseBackendName,
+  Traffic,
+} from "../../../gcp/apphosting.js";
+
+export const list_backends = tool(
+  {
+    name: "list_backends",
+    description:
+      "Retrieves a list of App Hosting backends in the current project. The `uri` is the public URL of the backend.",
+    inputSchema: z.object({
+      location: z
+        .string()
+        .optional()
+        .default("-")
+        .describe("Limit the listed backends to this region."),
+    }),
+    annotations: {
+      title: "List App Hosting backends.",
+      readOnlyHint: true,
+    },
+    _meta: {
+      requiresAuth: true,
+      requiresProject: true,
+    },
+  },
+  async ({ location } = {}, { projectId }) => {
+    if (!projectId) return NO_PROJECT_ERROR;
```

Comment:
This is already taken care of by the caller.

https://github.com/firebase/firebase-tools/blob/214d190ecd98844c1cb90d22afb51f584854775c/src/mcp/index.ts#L189

Reply:
Sure, but I still have to assert that projectId isn't undefined, otherwise I have to do checks all over the place. Does this hurt that much to bother?

Reply:
Yeah, it's a bit annoying that `projectId` is nullable even if `requiresProject` is true above.

Maybe we can do a clean up to supply a dummy `projectId: "missing-project"` if there there is no project present.